bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/LedgerDirsMonitor.java

```diff
@@ -40,117 +45,109 @@
 /**
  * Thread to monitor the disk space periodically.
  */
-class LedgerDirsMonitor extends BookieThread {
+class LedgerDirsMonitor {
     private static final Logger LOG = LoggerFactory.getLogger(LedgerDirsMonitor.class);
 
     private final int interval;
     private final ServerConfiguration conf;
     private final ConcurrentMap<File, Float> diskUsages;
     private final DiskChecker diskChecker;
     private final LedgerDirsManager ldm;
+    private ScheduledExecutorService executor;
+    private ScheduledFuture<?> checkTask;
 
-    public LedgerDirsMonitor(final ServerConfiguration conf, final DiskChecker diskChecker,
+    public LedgerDirsMonitor(final ServerConfiguration conf,
+                             final DiskChecker diskChecker,
                              final LedgerDirsManager ldm) {
-        super("LedgerDirsMonitorThread");
         this.interval = conf.getDiskCheckInterval();
         this.conf = conf;
         this.diskChecker = diskChecker;
         this.diskUsages = ldm.getDiskUsages();
         this.ldm = ldm;
     }
 
-    @Override
-    public void run() {
-        while (true) {
-            try {
-                List<File> writableDirs = ldm.getWritableLedgerDirs();
-                // Check all writable dirs disk space usage.
-                for (File dir : writableDirs) {
+    private void check() {
```

Comment:
Looks like a majority of the changes are because of removing the run() around the code and making it a check() method. I believe there is no logic change. If so, a comment will help the reviewer. I wish these diff tools are intelligent enough to identify things like that.

Reply:
yes there is no code change there. I will add a note to description and leave a comment so people will see it.